haystack/components/routers/conditional_router.py

```diff
@@ -136,11 +137,49 @@ def __init__(
         :param validate_output_type:
             Enable validation of routes' output.
             If a route output doesn't match the declared type a ValueError is raised running.
+        :param optional_variables:
+            A list of variable names that are optional in your route conditions and outputs.
+            If these variables are not provided at runtime, they will be set to `None`.
+            This allows you to write routes that can handle missing inputs gracefully without raising errors.
+
+            Example usage with a default fallback route:
+            ```python
+            routes = [
+                {
+                    "condition": '{{ path == "rag" }}',
+                    "output": "{{ question }}",
+                    "output_name": "rag_route",
+                    "output_type": str
+                },
+                {
+                    "condition": "{{ True }}",  # fallback route
+                    "output": "{{ question }}",
+                    "output_name": "default_route",
+                    "output_type": str
+                }
+            ]
+
+            router = ConditionalRouter(routes, optional_variables=["path"])
+
+            # When 'path' is provided, specific route is taken:
+            result = router.run(question="What?", path="rag")
+            assert result == {"rag_route": "What?"}
+
+            # When 'path' is not provided, fallback route is taken:
+            result = router.run(question="What?")
+            assert result == {"default_route": "What?"}
```

Comment:
Good example! Except this example actually works pre your changes. To get the error we were originally seeing you would need to put it in a Pipeline first. Even a single component pipeline would do. Otherwise it's a bit misleading here since this example would actually work without specifying the `optional_variables` parameter.